src/main/java/com/google/research/bleth/servlets/ListSimulationsServlet.java

```diff
@@ -34,11 +35,20 @@
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         Gson gson = new Gson(); // Used for json serialization.
 
-        // Read SimulationMetadata as a HashMap.
-        ImmutableMap<String, SimulationMetadata> simulations = SimulationMetadata.listSimulations();
+        // Read SimulationMetadata as a sorted LinkedHashMap.
+        Optional<String> sortProperty = Optional.empty();
+        Optional<Query.SortDirection> sortDirection = Optional.empty();
+        if (request.getParameter("sortProperty") != null) {
+            sortProperty = Optional.of(request.getParameter("sortProperty"));
+        }
+        if (request.getParameter("sortDirection") != null) {
```

Comment:
Having sortDirection set without a sortProperty is meaningless (and is likely a violation of this servlet's API).
Prefer to return an error if this is the case.

Reply:
Do we have to throw an exception in such case if we are using the `SortingParameters` you have suggested? I thought of constructing a `SortingParameters` optional only if both parameters are not null, and otherwise return the results unsorted (in practice, either both are provided or neither is provided).

Reply:
It's not a must, but - it is a better API. (In theory, if someone else were to write a new frontend for your service, they could be surprised at this behavior).
Your call.

Reply:
So this means we're treating differently each one of the four possible scenarios: (property, direction) X (provided, not provided), right?

Reply:
Possibly.

Or just - create the `SortingParameters` if both parameters are set, return an error if just one is set, return unsorted result if no parameters are set.